src/llama-kv-cells.h

```diff
@@ -9,6 +9,15 @@
 #include <set>
 #include <map>
 
+struct llama_kv_pos_mrope {
+    llama_pos y = 0;
+    llama_pos x = 0;
+    // return true if this position is greater than the other position
+    bool is_gt(const llama_kv_pos_mrope & other) const {
+        return (y > other.y) || (y == other.y && x > other.x);
+    }
+};
+
```

Comment:
Similarly, I think we can decouple the concept of M-RoPE here by declaring this struct to be more generic:

```c++
struct llama_kv_cell_ext {
    // 2D spatial positions, typically used for M-RoPE
    llama_pos x = 0;
    llama_pos y = 0;
};
```